packages/backend/src/helpers/generate-error-email.ts

```diff
@@ -0,0 +1,92 @@
+import { DateTime } from 'luxon'
+
+import { createRedisClient } from '@/config/redis'
+import logger from '@/helpers/logger'
+import { sendEmail } from '@/helpers/send-email'
+import Flow from '@/models/flow'
+
+const MAX_LENGTH = 80
+const CURR_DATETIME = DateTime.now()
+
+function truncateFlowName(flowName: string) {
+  return flowName.length > MAX_LENGTH
+    ? flowName.slice(0, MAX_LENGTH) + '...'
+    : flowName
+}
+
+export function createBodyErrorMessage(flowName: string): string {
+  const currDateTime = CURR_DATETIME.toFormat('MMM dd yyyy, HH:mm:ss a')
+  const bodyMessage = `
+    Dear user,
+    <br>
+    <br>
+    We have detected that your pipe: <strong>${flowName}</strong>, from plumber.gov.sg has failed as of ${currDateTime}.
+    <br>
+    Several scenarios could have caused this issue:
+    <ol>
+      <li>One of the apps you are using may be down.</li>
+      <li>Some parts of the pipe you have set up may not be working as intended.</li>
+    </ol>
+    <br>
+    Please check https://status.plumber.gov.sg/ to see if the status for Plumber is green or not. If one of the apps you are using are down, the plumber team would've already been alerted as well. 
+    <br>
+    <br>
+    If the pipe issue is urgent, please log in to plumber.gov.sg and check the pipe. Please retry the pipe under the executions tab subsequently. Otherwise, you may respond to this email and request assistance from the Plumber support team.
+    <br>
+    <br>
+    Regards,
+    <br>
+    Plumber Team
+  `
+  return bodyMessage
+}
+
+export async function sendErrorEmail(
+  flow: Flow,
+  userEmail: string,
+  isTestRun: boolean,
+) {
+  if (isTestRun) {
+    return
+  }
+
+  const truncatedFlowName = truncateFlowName(flow.name)
+  const redisClient = createRedisClient()
```

Comment:
Each redis client opens a persistent connection to redis. You shouldnt be creating one for each email but reuse the existing connection. There's a default limit of [10000 max connection](https://redis.io/docs/reference/clients/#accepting-client-connections)s (i think).

Also, use a different db for this use case. See config/redis, 
```
export const REDIS_DB_INDEX = {
  JOBS: 0,
  RATE_LIMIT: 1,
}
```

Reply:
oh i see, thks for the enlightenment sir, didnt know that it was the different database names